test_data/rules/iterable_contains_unrelated_type.dart

```diff
@@ -4,6 +4,8 @@
 
 // test w/ `dart test -N iterable_contains_unrelated_type`
 
+// @dart=2.19
```

Comment:
See also: https://github.com/dart-lang/linter/issues/4143

I'm not sure this is the right long term fix since it means we won't be testing the Dart 3.0 code path?

Reply:
The right long term fix is to delete the rule and test 😛 

For now I've referenced #4143 in the file. Good catch.